### PR TITLE
Leaderboard db

### PR DESCRIPTION
**Pull request by EliHaugu**
The pressing participants function is disabled for now since it's not needed yet as there are no participants/competitions :)

### CODE REVIEW

**Comment by EliHaugu** [comment]
noen som har noen ideer hvorfor det feilet checken? :) får ikke til med vercel :)

**Comment by EliHaugu** [comment]
Found one thing that made it fail and removed that, but now setup and yarn install failed, anyone know why or how I can try to test it again?

**Comment by SmithPeder** [comment]
I bet this failes becuase the `yarn.lock` is cracy here 😆 It's an 8000line diff, and you have only bumped react-markdown by x.x.2 patch. I would remove the package.json and yarn.lock changes from this PR 👍🏻

File: pages/api/leaderboardAPI/competition/[id].tsx [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  await mongoose.connect(url);
+
+  switch (method) {
+    case "GET": {
+      res
+        .status(200)
+        .json(
+          await ScoreModel.find({ CID: query })
+            .populate("UID")
+            .populate("CID")
```
Comment (SmithPeder) [inline pages/api/leaderboardAPI/competition/[id].tsx:21]: you can populate multiple with just `.populate("UID CID")`

File: pages/api/leaderboardAPI/participant/[id].tsx [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  await mongoose.connect(url);
+
+  switch (method) {
+    case "GET": {
+      res
+        .status(200)
+        .json(
+          await ScoreModel.find({ UID: query })
+            .populate("UID")
+            .populate("CID")
```
Comment (SmithPeder) [inline pages/api/leaderboardAPI/participant/[id].tsx:21]: same

File: tsconfig.json [hunk 3]
```diff
@@ -24,7 +24,7 @@
     "next-env.d.ts",
     "**/*.ts",
     "**/*.tsx",
-    "*.d.ts",
+    "*.d.ts", "models/leaderboardSchema.tsx", "models/leaderboardSchema.ts", "pages/api/leaderboardAPI/leaderboardAPI.tsx",
```
Comment (SmithPeder) [inline tsconfig.json:27]: Line 25 and 26 in this file already covers all `.tsx` and `.ts` files

**Comment by EliHaugu** [comment]
Thank you very much! Also see some lint issues here so will fix that

File: pages/leaderboard/index.tsx [hunk 4]
```diff
@@ -11,16 +11,31 @@ import TableRow from "@mui/material/TableRow";
...
+    "/api/leaderboardAPI/leaderboardMain",
+    fetcher,
+    {
+      refreshInterval: 5000,
+    }
+  );
+  const router = useRouter();
+  const navigateTo = function (to: string) {
+    router.push(to);
+  };
```
Comment (SmithPeder) [inline pages/leaderboard/index.tsx:31]: This function is kinda unnecessary, right? It is just a wrapper for `router.push()` with the same signature. So at best it allows you to write one less word? 😬

File: pages/api/leaderboardAPI/competition/[id].tsx (hunk 1 above)
Comment (EliHaugu) [reply pages/api/leaderboardAPI/competition/[id].tsx:21]: Thank you, didn't know that! :)

File: tsconfig.json (hunk 3 above)
Comment (EliHaugu) [reply tsconfig.json:27]: Don't know why that is added, will remove

File: pages/leaderboard/index.tsx (hunk 4 above)
Comment (EliHaugu) [reply pages/leaderboard/index.tsx]: Yeah thought the same but the tutorial did it so thought it had some purpose :sweat_smile: Thank you! Removed it for now as it's not used and lint is complaining

**Pull request merged**